Update local memory fmp and add asm op for absolute local addr

grjte opened the pull request:
This PR addresses #118 and #105.

- updates `fmp` max and min values for local memory addressing
- updates unit tests for `fmpadd` and `fmpupdate`
- adds new op `push.env.locaddr.i` to push the absolute address of the local at index `i` onto the stack
- updates unit & integration tests for env io ops
- refactors assembly op validation to a macro for greater flexibility and to reduce repeated code in op parsers

bobbinth reviewed (approved):
Looks great! And the macro is really awesome! Thank you.

I have just one small doc-related comment inline.

grjte commented on assembly/src/parsers/io_ops/env_ops.rs:
```diff
@@ -15,15 +17,18 @@ use super::{validate_op_len, AssemblyError, Operation, Token};
 /// input is unrecognized.
 pub fn parse_push_env(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
-    // validate the operation
-    validate_op_len(op, 3, 0, 0)?;
-    if op.parts()[1] != "env" {
-        return Err(AssemblyError::unexpected_token(op, "push.env.{var}"));
-    }
+    validate_operation!(op, "push.env.locaddr|sdepth");
```
Good catch, thanks! Fixed

grjte commented:
The previous version was missing 2 important validations:
- ensure the op is allowed in the context (i.e. num_proc_locals > 0)
- ensure the index of the local is valid

Sorry for the oversight 😬 I've updated it and added checks to the unit tests for these cases

bobbinth reviewed (approved):
Looks great! Thank you! And sorry I didn't notice the missing checks during my first review.
